src/codegen.cpp

```diff
@@ -4286,7 +4277,7 @@ static std::unique_ptr<Module> emit_function(jl_lambda_info_t *lam, jl_llvm_func
         #ifndef LLVM34
         SP = dbuilder.createFunction((DIDescriptor)dbuilder.getCU(),
         #else
-        SP = dbuilder.createFunction(CU,
+        SP = dbuilder.createFunction(topfile,
```

Comment:
I'm not sure this works. If the function isn't associated to a compile unit, it might not end up being represented in the binary.